tests/smoke_tests/backward_compat/test_backward_compat.py

```diff
@@ -223,9 +223,12 @@ def test_cluster_launch_and_exec(self, request, generic_cloud: str):
             need_launch_cmd = f'{self.ACTIVATE_CURRENT} && sky launch --infra {generic_cloud} -y -c {cluster_name}'
         commands = [
             f'{self.ACTIVATE_BASE} && {smoke_tests_utils.SKY_API_RESTART} && '
-            f'{self.ACTIVATE_CURRENT} && sky launch --infra {generic_cloud} -y {smoke_tests_utils.LOW_RESOURCE_ARG} --num-nodes 2 -c {cluster_name} examples/minimal.yaml',
-            f'{self.ACTIVATE_CURRENT} && sky autostop -i 10 -y {cluster_name}',
-            f'{self.ACTIVATE_CURRENT} && sky exec -d --infra {generic_cloud} --num-nodes 2 {cluster_name} sleep 120',
+            # Use --cloud since old version may not have --infra
+            f'sky launch --cloud {generic_cloud} -y {smoke_tests_utils.LOW_RESOURCE_ARG} --num-nodes 2 -c {cluster_name} examples/minimal.yaml',
+            f'{self.ACTIVATE_BASE} && sky autostop -i 10 -y {cluster_name}',
+            f'{self.ACTIVATE_BASE} && sky exec -d --infra {generic_cloud} --num-nodes 2 {cluster_name} sleep 120',
```

Comment:
I think this should be reverted to `--cloud` too

```suggestion
            f'{self.ACTIVATE_BASE} && sky exec -d --cloud {generic_cloud} --num-nodes 2 {cluster_name} sleep 120',
```